pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -0,0 +1,548 @@
+
+The `task` keyword that comes after the `blocks` property divides a YAML
+configuration file into major and distinct sections. Each `task` item can
+have multiple `jobs` items, an optional `prologue` section, an optional
+`epilogue` section as well as an optional `env_vars` block for defining
+environment variables and an optional `secrets` block for using existing
+`secrets` environment variables.
+
+#### jobs
+
+The `jobs` items of the `blocks` list allow you to define the commands that
+you want to execute.
+
+
+#### secrets
+
+The `secrets` property is used for using existing `secrets` environment
+variables.
+
+
+#### prologue
+
+As the `prologue` block is executed before a job, the result of the job that
+follows that `prologue` block has no effect on the execution of
+the `prologue` block.
+
+
+#### epilogue
+
+`epilogue` blocks are executed even if the pipeline jobs that precede them
+have failed.
+
+
+#### env_vars
+
+The elements of an `env_vars` array are name and value pairs that hold the name
+of the environment variable
+and the value of the environment variable, respectively.
+
+
+##### Example of env_vars
+
+    env_vars:
+      - name: VAR1
+         value: Environment Variable 1
+      - name: PI
+         value: "3.14159"
+
+The preceding YAML code exempt defines two environment variables named `VAR1`
+and `PI`. Both environment variables have string values, which means that
+numeric values that are not natural numbers need to be included in double
+quotes.
+
+
+### Example of task
+
+       task:
+          jobs:
+            - name: Execute hw.go
+              commands:
+                - echo $SEMAPHORE_PIPELINE_ID
+                - echo $HOME
+                - echo $SEMAPHORE_GIT_DIR
+                - pwd
+                - ls -al
+                - cat hw.go
+                - echo $VAR1
+                - echo $PI
+          prologue:
+              commands:
+                - checkout
+          env_vars:
+               - name: VAR1
+                 value: Environment Variable 1
+               - name: PI
+                 value: "3.14159"
+
+The indentation level of `prologue`, `epilogue`, `env_vars` and `jobs` properties should be the same.
+
+
+## Jobs
+
+The `jobs` items are essential for each Semaphore 2.0 pipeline because they
+allow you to define the actual commands that you want to execute.
+
+### name
+
+The value of the `name` property is a Unicode string that provides a
+description/name for a particular job. It is considered a good practice to give
+descriptive names to the jobs and the blocks items of a Semaphore 2.0 pipeline.
+
+
+### commands
+
+The `commands` property is an array of strings that holds the UNIX commands
+that will be executed for a job.
+
+
+#### The checkout command
+
+The `checkout` command is a special command offered by Semaphore 2.0 and is
+used for fetching the files of the GitHub repository into the virtual machine
+in order to be able to use them.
+
+
+### cmd_file
```

Comment:
cmd_file proparty is going to be renamed. Until that happens just remote all mentions of `cmd_file`